src/parse_nlp.jl

```diff
@@ -93,11 +93,33 @@ function _parse_NL_expr(m, x, tapevar, parent, values)
                 )
             else
                 opname = quot(x.args[1])
-                errorstring = "Unrecognized function \"$(x.args[1])\" used in nonlinear expression."
+                f = x.args[1]
+                errorstring = """
+                Unrecognized function \"$(f)\" used in nonlinear expression.
+
+                If the function exists, but is not within the scope of this call,
+                you should register it as a user-defined function before building
+                the model. For example:
+                ```julia
+                model = Model()
+                register(model, :$(f), 1, $(f), autodiff=true)
+                # ... variables and constraints ...
+                ```
+                """
                 errorstring2 = "Incorrect number of arguments for \"$(x.args[1])\" in nonlinear expression."
                 lookupcode = quote
                     if $(esc(m)).nlp_data === nothing
-                        error($errorstring)
+                        try
+                            register(
+                                $(esc(m)),
+                                $opname,
+                                1,
+                                $(esc(x.args[1]));
+                                autodiff = true,
+                            )
```

Comment:
Add a warning that the function is being auto-registered, and that we recommend you register this manually.